### PR TITLE
Problem 71: Some of the tests for Blending Editor fails

### PR DESCRIPTION
**Pull request by kegor**
Improvements related to fixing #71.

**Problem:**
Some of the tests for Blending Editor fails on TC (and sometimes locally).

**Solution:**
Tests were refactored a little, to be more similar with BlendingProvider tests.

**Tests:**
Finally, [tests on TC works with added BE tests](https://teamcity.starcounter.org/viewLog.html?buildId=46582&).

### CODE REVIEW

File: test/BlendingEditor.Tests/Test/LayoutGroupsPageTest.cs [hunk 1]
```diff
@@ -12,9 +10,11 @@ namespace BlendingEditor.Tests.Test
     //[Parallelizable(ParallelScope.Fixtures)]
     [TestFixture(Config.Browser.Chrome)]
     [TestFixture(Config.Browser.Edge)]
-    //[TestFixture(Config.Browser.Firefox)]
+    [TestFixture(Config.Browser.Firefox)]
     class LayoutGroupsPageTest : BaseTest
     {
+        private const int ScIncludeIndex = 0;
```
Comment (warpech) [inline test/BlendingEditor.Tests/Test/LayoutGroupsPageTest.cs]: All `starcounter-includes` should be given a slot attribute. Please just select the correct `starcounter-include` using `CssSelector("[slot='xxxxx'])").

File: test/BlendingEditor_TestHelper/Api/MainHandlers.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
+﻿using BlendingEditor_TestHelper.Helpers;
+using Starcounter;
+
+namespace BlendingEditor_TestHelper.Api
+{
+    public class MainHandlers
+    {
+        public void Register()
+        {
+            RegisterLyoutGroupsPageTestHandlers();
```
Comment (un-tone) [inline test/BlendingEditor_TestHelper/Api/MainHandlers.cs]: Typo: `RegisterLayout...`

File: test/BlendingEditor_TestHelper/Program.cs [hunk 3]
```diff
@@ -28,6 +30,7 @@ static void Main()
             Application.Current.Use(request =>
             {
                 // This is needed because this middlewear will be called on every request, also those for static files which shouldn't be handled.
+                // TODO : move file extensions into separate list
```
Comment (un-tone) [inline test/BlendingEditor_TestHelper/Program.cs]: Let's do it while we are here ;-)

File: test/BlendingEditor.Tests/Test/LayoutGroupsPageTest.cs [hunk 4]
```diff
@@ -10,123 +8,196 @@ namespace BlendingEditor.Tests.Test
 {
     // Those tests can NOT be run simultaneously, because changes that they make during execution interfere with each other
     //[Parallelizable(ParallelScope.Fixtures)]
-    [TestFixture(Config.Browser.Chrome)]
-    [TestFixture(Config.Browser.Edge)]
-    //[TestFixture(Config.Browser.Firefox)]
+    [TestFixture(Config.Browser.Chrome, ViewKeeperMode.EnableViewKeeper)]
+    [TestFixture(Config.Browser.Firefox, ViewKeeperMode.EnableViewKeeper)]
+    //[TestFixture(Config.Browser.Edge, ViewKeeperMode.EnableViewKeeper)]
```
[the remark this answers is not in the document]
Comment (kegor) [reply test/BlendingEditor.Tests/Test/LayoutGroupsPageTest.cs]: The comment was removed.

File: test/BlendingEditor_TestHelper/Api/MainHandlers.cs (hunk 2 above)
Comment (kegor) [reply test/BlendingEditor_TestHelper/Api/MainHandlers.cs]: The typo was fixed.

File: test/BlendingEditor_TestHelper/Program.cs (hunk 3 above)
Comment (kegor) [reply test/BlendingEditor_TestHelper/Program.cs]: Done.

File: test/BlendingEditor.Tests/Test/BaseTest.cs [hunk 5]
```diff
@@ -72,6 +74,7 @@ public void TestFixtureTearDown()
             }
 
             Driver?.Navigate().GoToUrl(Config.BlendingUrl + "/ResetData");
+            SetViewKeeper(true);
```
[the remark this answers is not in the document]
Comment (kegor) [reply test/BlendingEditor.Tests/Test/BaseTest.cs]: Done, one check for Driver == null was added in the top of TestFixtureTearDown method.

**Comment by warpech** [comment]
As it was discussed yesterday, ViewKeeper fixtures should be removed from this PR because ViewKeeper is tested in BlendingProvider.

**Comment by kegor** [comment]
@warpech, @PatrykSzwer, guys, latest TC build with added BlendingEditor parameters was successful (this is at least fifth successful TC build with BE-parameters in a row), please take a look at my latest iteration when you'll find a time and we will move forward with this PR.

**Comment by PatrykSzwer** [comment]
I had some troubles running those tests locally. When I tried to run them from VS level, there were some random problems - tests flakiness, TestHelper shutting down for no reason etc. But when I'm running tests trough test.bat It's working perfectly so I'll assume that's some kind of local problem. If @warpech won't have such problems and he approves, go ahead and merge this.

**Comment by kegor** [comment]
@PatrykSzwer I tried to use this case:

1. In Blending cmd: `git clean -xdf` —> `staradmin delete db --force` —> `Build`
2. In VS: Make BP startup project -> `Ctrl+F5`, same for BE, same for BE_TestHelper
3. Run All on BE.Tests

I tried these steps 3 times, and I only once saw one 'flaky' fail for only one test in FF. So maybe there are some differences in your local environment. But for sure we can ask someone to try reproduce yet.

**Comment by warpech** [comment]
@kegor please add BlendingEditor parameters to the TeamCity configuration of `Blending Master with Starcounter Develop (2.4)` after this is merged.

**Pull request merged**